### PR TITLE
Refactors logger file and adds several unit tests

### PR DESCRIPTION
**Pull request by Verlic**
This PR includes to major additions:

- Refactors the logger file and functions
- Configuration settings are now abstracted via an object

# Refactors logger file

This PR aims to increase code coverage for the `logger` functionality of the library and refactors the organization of the code to reduce the number of code lines per file. Each `stream` now lives in its own file.

## Configuration settings object

A new object is created when initializing the `agent`. The configuration object parses the package object and the environment variables sent as parameters and constructs a new Configuration object that is used to initialize instrumentation functions.

### CODE REVIEW

File: lib/configuration/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      channel: env.RELEASE_CHANNEL
+    };
+  },
+  /**
+   * @typedef {Object} Logger
+   * @property {Object} streams - An Array of Logger Stream configuration objects
+   */
+  logger: function (env) {
+    const streams = {};
+    const settings = require('./logger-settings');
```
Comment (elbuo8) [inline lib/configuration/index.js]: `require` should be at the top

File: lib/configuration/logger-settings.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+module.exports = {
```
Comment (elbuo8) [inline lib/configuration/logger-settings.js]: <3

File: lib/stubs.js [hunk 3]
```diff
@@ -29,6 +29,10 @@ const emptyLogger = {
   fatal: noop
 };
 
+var someObj = {
```
Comment (elbuo8) [inline lib/stubs.js]: ?

File: lib/tags.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  processTags(tags) {
+    if (typeof tags === 'object' && !Array.isArray(tags)) {
+      return this.extractTags(tags);
+    }
+
+    return Array.isArray(tags) ? tags : [];
+  }
+}
+
+module.exports = new TagUtils();
```
Comment (elbuo8) [inline lib/tags.js]: Why not just create static method and that way you don't have to call `new`

**Comment by nw** [comment]
Nice work, but obsolete. This is migrating to `logger` repo.

**Pull request closed without merging**